### PR TITLE
fixes

### PR DESCRIPTION
**Pull request by mandd**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/PostProcessorFunctions/riskMeasuresDiscrete.py [hunk 1]
```diff
@@ -212,14 +216,17 @@ def runStatic(self,inputDic, componentConfig=None):
         ## of this function.
         if componentConfig is None:
           inputName     = inp['name']
-          #inputDataIn   = inp['data']['input']
-          #inputDataOut  = inp['data']['output']
```
Comment (wangcj05) [inline framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: rm?

File: framework/PostProcessorFunctions/riskMeasuresDiscrete.py [hunk 2]
```diff
@@ -362,27 +370,27 @@ def runDynamic(self,inputDic,timeHistory):
...
       # Do not repeat the calculation if the system configuration is identical to the one of previous time instant
       if systemConfig == previousSystemConfig:
         for key in outputDic['data'].keys():
-          outputDic['data'][key][index] = outputDic['data'][key][index-1]
+          outputDic['data'][key][0][index] = outputDic['data'][key][0][index-1]
       else:
         staticOutputDic = self.runStatic(inputDic,systemConfig)
         for key in outputDic['data'].keys():
-          outputDic['data'][key][index] = staticOutputDic['data'][key]
+          outputDic['data'][key][0][index] = staticOutputDic['data'][key][0]
```
Comment (wangcj05) [inline framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: ```0``` index is used in several places, could you tell me why ```0``` index is always used, and why do we need this?

**Comment by wangcj05** [comment]
In addition, the test results should added in this PR

File: framework/PostProcessorFunctions/riskMeasuresDiscrete.py (hunk 1 above)
Comment (mandd) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: they are actually removed :-P

Comment (wangcj05) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: Got it.

File: framework/PostProcessorFunctions/riskMeasuresDiscrete.py (hunk 2 above)
Comment (wangcj05) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: I think you can either put a FIXME with the explanation and changed later, or you can fix it on this PR. Both are fine with me.

Comment (mandd) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: if it is ok to you I can address this when this PP will be moved in a PRA plugin

Comment (wangcj05) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: It is fine with me.

Comment (wangcj05) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: But I think we should add ```FIXME``` here.

Comment (mandd) [reply framework/PostProcessorFunctions/riskMeasuresDiscrete.py]: I just added some comments for the next round of development

**Comment by mandd** [comment]
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.failrunsMC | has changed outcome from <OK> to <FAILED> !!!!
**** Test framework.redundantInputs | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.redundantInputsSCgPC | has changed outcome from <OK> to <FAILED> !!!!
**** Test framework.redundantInputsSobol | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.test_xsd_input_fails | has changed outcome from <OK> to <SKIPPED> !!!!
**** Test framework/CodeInterfaceTests.genericInterfaceIO | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/CodeInterfaceTests.testMCMAMMOTHNoExecutable_RattlesnakeBison | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.RiskMeasuresDiscreteTimeDep | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.historySetSyncAll | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.historySetSyncMin | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.Grid | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.MonteCarlo | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.RomLoad | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.singleRunPlot | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.singleRunSubPlot | has changed outcome from <SKIPPED> to <FAILED> !!!!
*****************
**** THE FOLLOWING TESTS HAVE BEEN REMOVED IN 2ND FILE ****
**** Test framework/CodeInterfaceTests.DymolaTestTimeDepNoExecutableEnsembleModel | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestAliasAliasNoExecutable | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestNoExecutable | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/ROM/TimeSeries.ARMAreseedTest.......................................... | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.MonteCarlo skipped (skipped (Unable to import library: "PIL")) | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.Stratified skipped (skipped (Unable to import library: "PIL")) | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_IRR | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_NPV | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_NPVsearch | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_PI | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
*****************
**** THE FOLLOWING TESTS HAVE BEEN ADDED IN 2ND FILE ****
**** Test framework/CodeInterfaceTests.DymolaTestTimeDepNoExecutableEnsembleModel [incorrect test]skipped ( duplicated files specified: /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++1/dsin.txt /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++1/addinp.txt /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++5/dsin.txt /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++5/addinp.txt) | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/CodeInterfaceTests.RAVENrunningRAVEN | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestAliasAliasNoExecutable [incorrect test]skipped ( duplicated files specified: /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTestAlias/testDummyStep/1/snc01.i /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTestAlias/testDummyStep/2/snc01.i) | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestNoExecutable [incorrect test]skipped ( duplicated files specified: /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTest/testDummyStep/1/snc01.i /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTest/testDummyStep/2/snc01.i) | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/ROM/TimeSeries.ARMAreseedTest | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.MonteCarlo | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.Stratified | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test plugins/PRAplugin.TestGraphModel | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test plugins/PRAplugin/tests.TestGraphModel | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
*****************
Number of success first file:
431
Number of success second file:
435
NUMBER OF SUCCESSES INCREASED: +4
**** NUMBER OF TESTS IN THE TWO FILES ARE DIFFERENT: ****
**** first file :538 /= second file: 537****
TOTAL TIME OF TESTS INCREASED       : reference_time(908.6) < new_time(975.0)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(2.10812064965) v.s. new_time(2.24137931034)
*********************************

**Comment by mandd** [comment]
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.failrunsMC | has changed outcome from <OK> to <FAILED> !!!!
**** Test framework.redundantInputs | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.redundantInputsSCgPC | has changed outcome from <OK> to <FAILED> !!!!
**** Test framework.redundantInputsSobol | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.test_xsd_input_fails | has changed outcome from <OK> to <SKIPPED> !!!!
**** Test framework/CodeInterfaceTests.genericInterfaceIO | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/CodeInterfaceTests.testMCMAMMOTHNoExecutable_RattlesnakeBison | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.RiskMeasuresDiscreteTimeDep | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.historySetSyncAll | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.historySetSyncMin | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.Grid | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.MonteCarlo | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.RomLoad | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.singleRunPlot | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.singleRunSubPlot | has changed outcome from <SKIPPED> to <FAILED> !!!!
*****************
**** THE FOLLOWING TESTS HAVE BEEN REMOVED IN 2ND FILE ****
**** Test framework/CodeInterfaceTests.DymolaTestTimeDepNoExecutableEnsembleModel | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestAliasAliasNoExecutable | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestNoExecutable | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/ROM/TimeSeries.ARMAreseedTest.......................................... | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.MonteCarlo skipped (skipped (Unable to import library: "PIL")) | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.Stratified skipped (skipped (Unable to import library: "PIL")) | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_IRR | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_NPV | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_NPVsearch | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
**** Test plugins/CashFlow.CashFlow_PI | has been REMOVED! It is present in 1st file BUT NOT in 2nd file !!!!
*****************
**** THE FOLLOWING TESTS HAVE BEEN ADDED IN 2ND FILE ****
**** Test framework/CodeInterfaceTests.DymolaTestTimeDepNoExecutableEnsembleModel [incorrect test]skipped ( duplicated files specified: /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++1/dsin.txt /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++1/addinp.txt /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++5/dsin.txt /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/DMInterfaceTestTimedep/testDummyStep/BouncingBall++5/addinp.txt) | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/CodeInterfaceTests.RAVENrunningRAVEN | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestAliasAliasNoExecutable [incorrect test]skipped ( duplicated files specified: /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTestAlias/testDummyStep/1/snc01.i /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTestAlias/testDummyStep/2/snc01.i) | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/CodeInterfaceTests.RELAP5interfaceTestNoExecutable [incorrect test]skipped ( duplicated files specified: /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTest/testDummyStep/1/snc01.i /Users/mandd/projects/raven/tests/framework/CodeInterfaceTests/RELAP5interfaceTest/testDummyStep/2/snc01.i) | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/ROM/TimeSeries.ARMAreseedTest | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.MonteCarlo | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test framework/user_guide/ForwardSamplingStrategies.Stratified | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test plugins/PRAplugin.TestGraphModel | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
**** Test plugins/PRAplugin/tests.TestGraphModel | has been ADDED  ! It is present in 2nd file BUT NOT in 1st file !!!!
*****************
Number of success first file:
431
Number of success second file:
435
NUMBER OF SUCCESSES INCREASED: +4
**** NUMBER OF TESTS IN THE TWO FILES ARE DIFFERENT: ****
**** first file :538 /= second file: 537****
TOTAL TIME OF TESTS INCREASED       : reference_time(908.6) < new_time(975.0)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(2.10812064965) v.s. new_time(2.24137931034)
*********************************

**Comment by wangcj05** [comment]
Library issues, should be fixed later:
```
**** Test framework/user_guide/ravenTutorial.MonteCarlo | has changed outcome from <SKIPPED> to <OK> !!!!
**** Test framework/user_guide/ravenTutorial.RomLoad | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.singleRunPlot | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/user_guide/ravenTutorial.singleRunSubPlot | has changed outcome from <SKIPPED> to <FAILED> !!!!
*****************
```

**Pull request merged**